com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs

```diff
@@ -570,106 +639,158 @@ public override void ProcessInteractable(XRInteractionUpdateOrder.UpdatePhase up
-                            {
-                                targetTransform.Scale = ManipulationLogic.scaleLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
-                            }
+                            targetTransform.Scale = ManipulationLogic.scaleLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
                         }
+                    }
 
-                        using (RotateLogicMarker.Auto())
+                    using (RotateLogicMarker.Auto())
+                    {
+                        if (allowedManipulations.IsMaskSet(TransformFlags.Rotate))
                         {
-                            if (allowedManipulations.IsMaskSet(TransformFlags.Rotate))
-                            {
-                                targetTransform.Rotation = ManipulationLogic.rotateLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
-                            }
+                            targetTransform.Rotation = ManipulationLogic.rotateLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
                         }
+                    }
 
-                        using (MoveLogicMarker.Auto())
+                    using (MoveLogicMarker.Auto())
+                    {
+                        if (allowedManipulations.IsMaskSet(TransformFlags.Move))
                         {
-                            if (allowedManipulations.IsMaskSet(TransformFlags.Move))
-                            {
-                                targetTransform.Position = ManipulationLogic.moveLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
-                            }
+                            targetTransform.Position = ManipulationLogic.moveLogic.Update(interactorsSelecting, this, targetTransform, useCenteredAnchor);
                         }
-
-                        ApplyTargetPose(targetTransform);
                     }
+
+                    ApplyTargetTransform();
+                }
+                else if (useForces && updatePhase == XRInteractionUpdateOrder.UpdatePhase.Fixed)
+                {
+                    ApplyForcesToRigidbody();
                 }
             }
         }
 
         /// <summary>
-        /// Once the <paramref name="targetPose"/> has been determined, this method is called
+        /// Once the <paramref name="targetTransform"/> has been determined, this method is called
         /// to apply the target pose to the object. Calls <see cref="ModifyTargetPose"/> before
         /// applying, to adjust the pose with smoothing, constraints, etc.
         /// </summary>
-        /// <param name="targetPose">
-        /// The target position, rotation, and scale to set the object to.
-        /// <param/>
-        private void ApplyTargetPose(MixedRealityTransform targetPose)
+        private void ApplyTargetTransform()
```

Comment:
Out of curiosity, why did this method change to use a cached local value? I'm generally not a huge fan of methods that rely on some state elsewhere being modified first (as opposed to explicitly passing the latest state in), so I'm curious about the motivation here 😅

Reply:
The value needs to be stored as member variable because it is set in Update and read in FixedUpdate. Once it is a member variable, I do not see any value in passing it as argument once more.

Reply:
Maybe I'm reading this wrong but...are you sure (for this method)? It still looks like `ApplyTargetTransform` is called at the same time it used to be, which is inside a `if (updatePhase == XRInteractionUpdateOrder.UpdatePhase.Dynamic)` block.

Reply:
Additionally, could we just read `HostTransform.position` and `HostTransform.rotation` directly in `ApplyForcesToRigidbody` instead of caching the `targetTransform` just to only use two component parts of it?

Reply:
Oh I see: this method also persists some modifications via `ModifyTargetPose`